servant-client/src/Servant/Common/Req.hs

```diff
@@ -214,6 +215,10 @@ instance MonadBaseControl IO ClientM where
   -- restoreM :: StM ClientM a -> ClientM a
   restoreM st = ClientM (restoreM st)
 
+-- | Try clients in order, last error is preserved.
+instance Semigroup (ClientM a) where
+  a <> b = a `catchError` \_ -> b
```

Comment:
An alternative to using `Alt` would be to add a custom combinator. E.g.:

```haskell
-- | Try clients in order, last error is preserved.
orClient :: ClientM a -> ClientM a -> ClientM a
orClient a b = a `catchError` \_ -> b
```

I would prefer this over a wrong `Semigroup` instance.